crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -121,15 +123,47 @@ impl Plugin for VisibilityPlugin {
 pub fn calculate_bounds(
     mut commands: Commands,
     meshes: Res<Assets<Mesh>>,
-    without_aabb: Query<(Entity, &Handle<Mesh>), (Without<Aabb>, Without<NoFrustumCulling>)>,
+    without_aabb_or_with_changed_mesh: Query<
+        (Entity, &Handle<Mesh>),
+        (
+            Or<(Without<Aabb>, Changed<Handle<Mesh>>)>,
+            Without<NoFrustumCulling>,
+        ),
+    >,
+    mut entity_mesh_map: Local<HashMap<Handle<Mesh>, Vec<Entity>>>,
+    mut mesh_events: EventReader<AssetEvent<Mesh>>,
 ) {
-    for (entity, mesh_handle) in without_aabb.iter() {
+    for (entity, mesh_handle) in without_aabb_or_with_changed_mesh.iter() {
+        // Record entities that have mesh handles.
+        entity_mesh_map
+            .entry(mesh_handle.clone())
+            .or_default()
+            .push(entity);
```

Comment:
Like I said in the main thread, I think this is the right tradeoff. We should leave a comment documenting the edge case though.

Reply:
Documented in 7b20f607f6eba53fcc3a417f691d167b90e22f44.

Reply:
Note that in b02be5c18b8f29a6ac8aac6bda3dc466dffed11a I deleted the comment. I believe that, the way `register` and `deregister` are used, this is no longer possible. It may even be better if I move `deregister` to be the first thing before `register` so it can't be used wrong (but would also involve needless calls to `deregister` when registering an entity<->mesh for the first time). What do y'all think?

Reply:
Also, if it _is_ still possible, then we probably want `deregister` to do a `while let Some(idx) = entities.position()` so that we delete all occurrences.